src/hooks.tsx

```diff
@@ -119,6 +118,20 @@ export function useTabNameContext(): TabName {
   return c
 }
 
+export function useLayoutHeight(initialHeight: number = 0) {
+  const [height, setHeight] = useState(initialHeight)
+
+  const getHeight = useCallback(
+    (event: LayoutChangeEvent) => {
+      const latestHeight = event.nativeEvent.layout.height
+      if (latestHeight !== height) {
+        setHeight(latestHeight)
```

Comment:
Is this check necessary? doesn't react not even re-render if the value is the same?

I believe this is actually an anti-pattern, since now `getHeight` relies on the `height` variable, causing it to update its reference based on its parent state when this isn't necessary.

For example, you could just dispatch it with a callback too, right?

```tsx
setHeight(height => height === latestHeight ? height : latestHeight)
```

However, that becomes the same as just doing `setHeight(latestHeight)`

Reply:
Maybe relevant: https://github.com/facebook/react/issues/28779

Reply:
Also see:
https://legacy.reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update

> Bailing out of a state update
If you update a State Hook to the same value as the current state, React will bail out without rendering the children or firing effects. (React uses the [Object.is comparison algorithm](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/is#Description).)

> Note that React may still need to render that specific component again before bailing out. That shouldn’t be a concern because React won’t unnecessarily go “deeper” into the tree. If you’re doing expensive calculations while rendering, you can optimize them with useMemo.

The problem here is that all the code in the component will run again unnecessarily, even though it doesn't go "deeper" or actually render anything. By preventing setting the state, we can optimize against that.